### PR TITLE
add dedication contact details to receipt emails

### PR DESCRIPTION
**Pull request by hejkal**
**NOTE: DO NOT discuss internal CommitChange information in your PR; this PR will be public.
Link back to the issue in the Tix repo when you need to do that.**

closes https://github.com/CommitChange/tix/issues/4110

### CODE REVIEW

**Review by wwahammy — APPROVED** [review]
Looks good!

**Comment by wwahammy** [comment]
@hejkal the tests seem to have failed though. Otherwise it looked good! :laughing:

**Comment by hejkal** [comment]
> @hejkal the tests seem to have failed though. Otherwise it looked good! 😆

@wwahammy Yep, there are tests failing that I thought weren't related, trying to figure that out

File: app/views/donation_mailer/_donation_payment_table.html.erb [hunk 1]
```diff
@@ -84,23 +84,23 @@
 		<% @contact = JSON.parse(donation.dedication).dig('contact') %>
 		<tr>
 			<td><strong>Acknowledgement Email</strong></td>
-			<% if @contact['email'] %>
+			<% if @contact && @contact['email'] %>
```
Comment (wwahammy) [inline app/views/donation_mailer/_donation_payment_table.html.erb]: Does this work for all of these? It's a bit cleaner. If so, could you use it for the similar ones here?

```suggestion
			<% if @contact&.dig('email') %>
```

**Review by wwahammy — COMMENTED** [review]
I suggested an idea that I think will clean this up a bit but YMMV.

File: app/views/donation_mailer/_donation_payment_table.html.erb (hunk 1 above)
Comment (hejkal) [reply app/views/donation_mailer/_donation_payment_table.html.erb]: yep, looks like that works, thanks!

**Review by wwahammy — APPROVED** [review]

**Comment by wwahammy** [comment]
Looks good @hejkal!

**Pull request merged**